src/fifelse.c

```diff
@@ -140,3 +140,182 @@ SEXP fifelseR(SEXP l, SEXP a, SEXP b, SEXP na) {
   UNPROTECT(nprotect);
   return ans;
 }
+
+SEXP fcaseR(SEXP args) {
+  int n=length(args)-2;
+  if (n % 2) {
```

Comment:
I guess the first element of `args` is a throwaway? Otherwise I'd expect `n` to be odd if including the `default`

Reply:
The first element of `args` is `default`. Look how the function is defined in the wrappers.R file. The last element of `args` is `R_NilValue`. This is my understanding after reading this :
[https://marlin-na.github.io/r-api/pairlists/](https://marlin-na.github.io/r-api/pairlists/)

Reply:
oh it's the last arg that's the difference. I thought the first was throwaway because of CDR right away. also didn't expect the R_NilValue to count towards length but I guess it must. thanks for clarifying